src/martialarts.cpp

```diff
@@ -74,6 +68,46 @@ void add_if_exists( JsonObject &jo, Container &cont, bool was_loaded,
     }
 }
 
+class ma_skill_reader : public generic_typed_reader<ma_skill_reader>
+{
+    public:
+        std::pair<skill_id, int> get_next( JsonIn &jin ) const {
+            JsonObject jo = jin.get_object();
+            return std::pair<skill_id, int>( skill_id( jo.get_string( "name" ) ), jo.get_int( "level" ) );
+        }
+        template<typename C>
+        void erase_next( JsonIn &jin, C &container ) const {
+            const skill_id id = skill_id( jin.get_string() );
+            reader_detail::handler<C>().erase_if( container, [&id]( const std::pair<skill_id, int> &e ) {
+                return e.first == id;
+            } );
+        }
+};
+
+class ma_weapon_damage_reader : public generic_typed_reader<ma_weapon_damage_reader>
+{
+    public:
+        std::map<std::string, damage_type> dt_map = get_dt_map();
```

Comment:
Why store a local copy of that map? The map will never change and the copy is not changed either, so why make a copy?